extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/BasicHTTPEscalator.java

```diff
@@ -19,49 +19,143 @@
 
 package org.apache.druid.security.basic.authentication;
 
+import com.fasterxml.jackson.annotation.JacksonInject;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.inject.Provider;
+import org.apache.druid.java.util.common.logger.Logger;
 import org.apache.druid.java.util.http.client.CredentialedHttpClient;
 import org.apache.druid.java.util.http.client.HttpClient;
-import org.apache.druid.java.util.http.client.auth.BasicCredentials;
+import org.apache.druid.java.util.http.client.Request;
+import org.apache.druid.java.util.http.client.auth.Credentials;
 import org.apache.druid.metadata.PasswordProvider;
+import org.apache.druid.security.basic.BasicAuthDBConfig;
+import org.apache.druid.security.basic.escalator.db.cache.BasicEscalatorCacheManager;
+import org.apache.druid.security.basic.escalator.entity.BasicEscalatorCredential;
 import org.apache.druid.server.security.AuthenticationResult;
 import org.apache.druid.server.security.Escalator;
 
+import java.util.concurrent.atomic.AtomicReference;
+
 @JsonTypeName("basic")
 public class BasicHTTPEscalator implements Escalator
 {
-  private final String internalClientUsername;
-  private final PasswordProvider internalClientPassword;
+  private static final Logger LOG = new Logger(BasicHTTPEscalator.class);
+  public static final int DEFAULT_INTERNAL_CLIENT_CREDENTIAL_POLL_SECONDS = 10;
+
+  private final Provider<BasicEscalatorCacheManager> cacheManager;
   private final String authorizerName;
+  private final BasicAuthDBConfig dbConfig;
+  private final AtomicReference<BasicEscalatorCredential> cachedEscalatorCredential;
+  private final AtomicReference<Long> lastVerified;
+  private final int internalClientCredentialPoll;
 
   @JsonCreator
   public BasicHTTPEscalator(
+      @JacksonInject Provider<BasicEscalatorCacheManager> cacheManager,
       @JsonProperty("authorizerName") String authorizerName,
       @JsonProperty("internalClientUsername") String internalClientUsername,
-      @JsonProperty("internalClientPassword") PasswordProvider internalClientPassword
+      @JsonProperty("internalClientPassword") PasswordProvider internalClientPassword,
+      @JsonProperty("enableCacheNotifications") Boolean enableCacheNotifications,
+      @JsonProperty("cacheNotificationTimeout") Long cacheNotificationTimeout,
+      @JsonProperty("internalClientCredentialPoll") Integer internalClientCredentialPoll
   )
   {
+    this.cacheManager = cacheManager;
     this.authorizerName = authorizerName;
-    this.internalClientUsername = internalClientUsername;
-    this.internalClientPassword = internalClientPassword;
+    this.cachedEscalatorCredential = new AtomicReference<>(
+        new BasicEscalatorCredential(internalClientUsername, internalClientPassword.getPassword())
+    );
+    this.dbConfig = new BasicAuthDBConfig(
```

Comment:
do we really need this dbConfig inside escalator ? 
There are many places where i notice new BasicDBConfig instances being created at many places and most of the ppts being passed are null, causes confusion. If we can avoid it or inject the configs via guice would be helpful.

Reply:
Not anymore, I added this to be able to support escalator rolling credential update within CoordinatorBasicEscalatorCacheNotifier. Since i'm removing that feature altogether in the PR, this will get cleaned up

Reply:
Fixed, cleaned up dbConfig usage inside escalator